src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIRequestSmelter.java

```diff
@@ -448,7 +448,7 @@ protected BlockPos getPositionOfOvenToRetrieveFuelFrom()
             {
                 final FurnaceTileEntity furnace = (FurnaceTileEntity) entity;
 
-                if (furnace.getStackInSlot(FUEL_SLOT).getItem() == Items.BUCKET)
+                if (!compareItemStackListIgnoreStackSize(getOwnBuilding().getAllowedFuel(), furnace.getStackInSlot(FUEL_SLOT), false, false) && !furnace.getStackInSlot(FUEL_SLOT).isEmpty())
```

Comment:
Can you do the not empty first? It's a cheaper check, and there is no reason to do the list comparison if the slot is empty

Reply:
Yep good point. Done 👍